IS1001Test.py

```diff
@@ -12,12 +12,356 @@
+                'description': 'A URI identifying a human-readable web page with information about the error'
+              },
+              'error_description': {
+                'type': 'string',
+                'description': 'Human-readable ASCII text providing additional information'
+              }
+            },
+            'required': ['error'],
+            'type': 'object',
+            'minItems': 1,
+            'description': 'Object defining error type and description',
+          }
+        },
+      }],
+    }
+  }
+
+  Example "apis" Object:
+  {
+      'auth': {
+          'raml': 'AuthorizationAPI.raml',
+          'version': 'v1.0',
+          'hostname': 'example.co.uk',
+          'port': 80,
+          'spec_branch': 'v1.0-dev',
+          'spec_path': 'cache/is-10',
+          'base_url': 'http://example.co.uk:80',
+          'name': 'Authorization API',
+          'ip': '127.0.1.1',
+          'url': 'http://example.co.uk:80/x-nmos/auth/v1.0/',
+          'spec': <Specification.Specification object at 0x7f5b3b0719b0>
+      }
+  }
+  """
+
     def __init__(self, apis):
         GenericTest.__init__(self, apis)
+
+        self.url = self.apis[AUTH_API_KEY]["url"]
+        self.bearer_token = None
+        self.client_data = None
+
+    def set_up_tests(self):
+        try:
+            signup_data = {
+                'username': AUTH_USERNAME,
+                'password': AUTH_PASSWORD,
+                'is04': 'read',
+                'is05': 'write'
+            }
+            status, response = self.do_request(
+                method="POST", url=self.url + 'signup', data=signup_data
+            )
+
+            if status is not True:
+                raise Exception
+
+        except Exception:
+            print("""
+                Ensure a User is already set-up on the Authorization Server that corresponds
+                to the 'AUTH_USERNAME' and 'AUTH_PASSWORD' config options
+            """)
+
+    def tear_down_tests(self):
+        try:
+            self._make_auth_request(method="GET", url_path='delete_client/' + self.client_data.client_id)
+        except Exception:
+            pass
+
+    def _make_auth_request(self, method, url_path, data=None, auth=(AUTH_USERNAME, AUTH_PASSWORD)):
+        """Utility option for making requests with Basic Authorization"""
+        return self.do_request(
+            method=method, url=self.url + url_path, data=data, auth=auth
+        )
+
+    def test_01_register_user(self, test):
+        """Test registering a client to the `register_client` endpoint"""
+
+        RECOMMENDED_RESPONSE_FIELDS = ["client_secret", "redirect_uris"]
+        ARRAY_RESPONSE_FIELDS = ["redirect_uris", "grant_types", "response_types"]
+        LIST_DELIMITER = '\n'
+
+        with open("test_data/IS1001/register_client_request_data.json") as resource_data:
+            request_data = json.load(resource_data)
+
+        status, response = self._make_auth_request(

```

Comment:
Is this somewhat a pre-requisite for the other tests? If I run only test_02 will it work? If necessary it might be worth splitting this into a method which can be called before each test runs, much like 'do_registry_basics_prereqs' in IS0401.